### PR TITLE
Send two chunks of lookup data

### LINKED ISSUE #3744: Send two chunks from the launcher

### PR DESCRIPTION
**Pull request by mariaschett**
Fixes #3744 .

### CODE REVIEW

File: oak_functions_launcher/src/lookup.rs [hunk 1]
```diff
@@ -16,42 +16,98 @@
...
 use prost::Message;
-use std::{fs, path::PathBuf};
+use std::{fs, path::PathBuf, vec::IntoIter};
 use ubyte::ByteUnit;
 
+struct UpdateClient<'a> {
+    inner: &'a mut OakFunctionsAsyncClient<ConnectorHandle>,
+}
+
+impl<'a> UpdateClient<'_> {
```
Comment (andrisaar) [inline oak_functions_launcher/src/lookup.rs:31]: do you have to give a name to the first lifetime parameter here? I don't think you're explicitly referencit it anywhere.

**Review by andrisaar — APPROVED** [review]

**Review by tiziano88 — APPROVED** [review]

File: oak_functions_launcher/src/lookup.rs (hunk 1 above)
Comment (mariaschett) [reply oak_functions_launcher/src/lookup.rs]: No, I don't, thanks!

File: oak_functions_launcher/src/lookup.rs [hunk 2]
```diff
@@ -16,42 +16,98 @@
...
+        .next()
+        .expect("Chunk lookup data should have returned the empty lookup data.");
+    let next_chunk = chunks.next();
 
-    match client.update_lookup_data(&request).await {
-        Ok(_) => Ok(()),
-        Err(err) => Err(anyhow!("couldn't send lookup data: {:?}", err)),
+    match next_chunk {
+        None => update_client.start_and_finish(current_chunk).await,
+        Some(_) => update_client.start(chunks, current_chunk, next_chunk).await,
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply oak_functions_launcher/src/lookup.rs:110]: `start` and `start_and_finish` are already called as I intended.

The plan would be to call `continue_`  at the end of `start`.

```
    #[async_recursion]
    async fn continue_(
         mut chunks: IntoIter<LookupDataChunk>,
        current_chunk: LookupDataChunk,
        next_chunk: Option<LookupDataChunk>,
    ) -> anyhow::Result<()> 
  // call finish when iterator is empty.
```
However, I agree, I am not too happy of passing the chunks around, too.

I'll see whether it is better to store `chunks` in the `UpdateClient` and use `peek`.

File: oak_functions_launcher/tests/integration_test.rs [hunk 3]
```diff
@@ -123,65 +123,56 @@ async fn test_load_large_lookup_data() {
...
-    let max_chunk_size = ByteUnit::Kibibyte(2);
-    let entry_size = ByteUnit::Byte(10);
-    let entries_count = (max_chunk_size / entry_size).as_u64() as u32;
-    let entries = oak_functions_test_utils::create_test_lookup_data(entry_size, 0, entries_count);
+    let max_chunk_size = ByteUnit::Kilobyte(2);
 
-    let lookup_data_file = oak_functions_test_utils::write_to_temp_file(
-        &oak_functions_test_utils::serialize_entries(entries.clone()),
+    // Initialize with 1 chunk.
+    let entries_1chunk = oak_functions_test_utils::create_test_lookup_data(max_chunk_size, 1);
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply oak_functions_launcher/tests/integration_test.rs]: You are right: changed to `entries_one_chunks`

File: oak_functions_launcher/tests/integration_test.rs [hunk 4]
```diff
@@ -123,65 +123,56 @@ async fn test_load_large_lookup_data() {
...
 
-    let response = client
-        .invoke(&invoke_request)
-        .await
-        .expect("Failed to receive response.");
-
-    assert!(response.is_ok());
-    assert_eq!(value2.to_owned(), response.unwrap().body);
+    // Write 2 chunks in lookup data.
+    let enteries_2chunks = oak_functions_test_utils::create_test_lookup_data(max_chunk_size, 2);
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply oak_functions_launcher/tests/integration_test.rs]: Done.

File: oak_functions_test_utils/src/lib.rs [hunk 5]
```diff
@@ -85,21 +85,24 @@ pub fn serialize_entries(entries: HashMap<Vec<u8>, Vec<u8>>) -> Vec<u8> {
...
-    end: u32,
+    max_chunk_size: ByteUnit,
+    chunk_count: u32,
 ) -> HashMap<Vec<u8>, Vec<u8>> {
     let mut entries = std::collections::HashMap::new();
-    let entry_size = entry_size.as_u64() as usize;
 
-    let key_prefix = vec![0u8; (entry_size / 2) - 4];
+    let entry_size = ByteUnit::Byte(10);
+    // This has to be consistent with the overhead set in chunking up lookup data.
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply oak_functions_test_utils/src/lib.rs:97]: Ah, I see what you mean, but I am not sure a const is worth it. 

I could move `create_test_lookup_data` back to the integration test and expose a `OVERESTIMATED_DELIMITER_SIZE` and  `ESTIMATED_CHUNK_OVERHEAD`  from the launcher.

However, I think it is worth more having this in `test_utils`.

Let me know if you'd like me to change this!

**Review by mariaschett — COMMENTED** [review]
Thank you for the reviews!

**Pull request merged**